packages/utils/test/composeAbortSignals.test.ts

```diff
@@ -0,0 +1,109 @@
+import { composeAbortSignals } from '../src/composeAbortSignals'
+import { range } from 'lodash'
+import fetch from 'node-fetch'
+
+describe(composeAbortSignals, () => {
+    let controllers: AbortController[]
+    let composedSignal: AbortSignal
+
+    describe('given zero pre-aborted signals', () => {
+        beforeEach(() => {
+            controllers = range(10).map(() => new AbortController())
+            composedSignal = composeAbortSignals(...controllers.map((c) => c.signal))
+        })
+
+        it('initial state is aborted=false', () => {
+            expect(composedSignal.aborted).toBeFalse()
+        })
+
+        it('transitions to aborted=true when a signal aborts', () => {
+            controllers[4].abort()
+            expect(composedSignal.aborted).toBeTrue()
+        })
+
+        it('event "abort" is emitted when a signal aborts', () => {
+            const listener = jest.fn()
+            composedSignal.addEventListener('abort', listener)
+
+            expect(listener).toHaveBeenCalledTimes(0)
+            controllers[6].abort()
+            expect(listener).toHaveBeenCalledTimes(1)
+            expect(listener.mock.calls[0][0].type).toEqual('abort')
+        })
+
+        it('event "abort" is emitted max once', () => {
+            const listener = jest.fn()
+            composedSignal.addEventListener('abort', listener)
+            controllers[3].abort()
+            controllers[6].abort()
+            expect(listener).toHaveBeenCalledTimes(1)
+        })
+
+        it('onabort if set is invoked when a signal aborts', () => {
+            const listener = jest.fn()
+            ;(composedSignal as any).onabort = listener
```

Comment:
Comment about why casting is needed? (also line 54)

Maybe some rewording for the test name?

Reply:
First part fixed, I think name is fine?

Reply:
Does it mean something like "if `onbort` listener is set, it is invoked when a signal aborts?"